lib/std/fmt.zig

```diff
@@ -976,8 +976,7 @@ pub fn formatUnicodeCodepoint(
     var buf: [4]u8 = undefined;
     const len = unicode.utf8Encode(c, &buf) catch |err| switch (err) {
         error.Utf8CannotEncodeSurrogateHalf, error.CodepointTooLarge => {
-            const len = unicode.utf8Encode(unicode.replacement_character, &buf) catch unreachable;
-            return formatBuf(buf[0..len], options, writer);
+            return formatBuf(unicode.utf8EncodeComptime(unicode.replacement_character), options, writer);
```

Comment:
Almost, needs to be `formatBuf(&unicode.utf8EncodeComptime(unicode.replacement_character), options, writer);` to pass the returned array as a slice.

Reply:
> Note that you can run just this test, for example, with the command zig test lib/std/std.zig --test-filter "utf8 encode comptime".

I can't hide it anymore 😄😄, I still didn't manage to compile zig from source on my m2 macbook (llvm keeps erroring). So I'm only relying on the GH action.

Reply:
but it should be ok for the compiler to pass a string as oposed to a slice to a `[]const u8` right?

Reply:
Taking the address of, say, a returned `[3]u8` gives you a `*const [3]u8`.  The type of a string literal is, say, `*const [3:0]u8`.  Both of these types coerce to a slice directly without any casting.

Reply:
ok but `[3]u8` does not coerce to a slice without address-of

Reply:
> ok but `[3]u8` does not coerce to a slice without address-of

Correct, a slice is a kind of pointer in Zig, and can only implicitly coerced to from other pointers, e.g. `@as([]const i32, &.{1, 2, 3})` is valid but `@as([]const i32, .{1, 2, 3})` is invalid.